src/main/java/uk/gov/hmcts/reform/sscs/service/HearingsMapping.java

```diff
@@ -0,0 +1,255 @@
+package uk.gov.hmcts.reform.sscs.service;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.model.HearingWrapper;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.NO;
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.YES;
+
+@SuppressWarnings({"PMD.LinguisticNaming","PMD.UnnecessaryLocalBeforeReturn"})
+// TODO Unsuppress in future
+public final class HearingsMapping {
+
+    @Value("${exui.url}")
+    private static String exUiUrl;
+
+    private HearingsMapping() {
+
+    }
+
+    public static void updateFlags(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+        caseData.setAutoListFlag(shouldBeAutoListed(caseData.getAutoListFlag()));
+        caseData.setHearingsInWelshFlag(shouldBeHearingsInWelshFlag(caseData.getHearingsInWelshFlag()));
+        caseData.setAdditionalSecurityFlag(shouldBeAdditionalSecurityFlag(caseData.getAdditionalSecurityFlag()));
+        caseData.setSensitiveFlag(shouldBeSensitiveFlag(caseData.getSensitiveFlag()));
+    }
+
+    public static HmcCaseDetails createHmcCaseDetails(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+
+        return HmcCaseDetails.builder()
+                .caseDeepLink(getCaseDeepLink(caseData.getCcdCaseId()))
+                .caseManagementLocationCode(getCaseManagementLocationCode(caseData.getCaseManagementLocation()))
+                .build();
+    }
+
+    public static HearingRequest createHearingRequest(HearingWrapper wrapper) {
+        SscsCaseData caseData = wrapper.getUpdatedCaseData();
+
+        HearingRequest hearingRequest = HearingRequest.builder().build();
```

Comment:
Can you not use the builder methods to set these fields?

Reply:
Originally I did have other logic that prevented that/made it tidy, I changed it but didn't change this back, ill do this now